File: sdk-java/src/main/java/io/littlehorse/sdk/worker/CheckpointContext.java
package io.littlehorse.sdk.worker;

/**
 * Mutable context object passed to checkpointable functions for emitting log output.
 */
public class CheckpointContext {
    private String logOutput;

    /**
     * Creates an empty checkpoint context.
     */
    public CheckpointContext() {
        this.logOutput = "";
    }

    /**
     * Appends a value to the checkpoint log output.
     *
     * @param thing value to append; null values are appended as the literal string {@code null}
     */
    public void log(Object thing) {
        if (thing != null) {
            logOutput += thing.toString();
        } else {
            logOutput += "null";
        }
    }

    /**
     * Returns all accumulated checkpoint log output.
     *
     * @return concatenated checkpoint log output
     */
    public String getLogOutput() {
        return logOutput;
    }
}

File: sdk-go/littlehorse/checkpoint_context.go
package littlehorse

import "fmt"

// CheckpointContext contains runtime information for a checkpoint execution.
// It provides logging capabilities that are specific to the checkpoint being executed.
type CheckpointContext struct {
	logOutput string
}

// NewCheckpointContext creates a new CheckpointContext instance.
func NewCheckpointContext() *CheckpointContext {
	return &CheckpointContext{
		logOutput: "",
	}
}

// Log appends the string representation of the provided item to the checkpoint's log output.
// If the item is nil, "nil" is appended to the log.
func (cc *CheckpointContext) Log(item interface{}) {
	if item != nil {
		cc.logOutput += fmt.Sprint(item)
	} else {
		cc.logOutput += "nil"
	}
}

// GetLogOutput returns the accumulated log output for this checkpoint.
func (cc *CheckpointContext) GetLogOutput() string {
	return cc.logOutput
}
